### DISCUSSION TITLE
Using QSelect to make a filter menu with the input within the menu

### DISCUSSION THREAD

**Discussion by acordy** [quasarframework/quasar#17637] [General - Components / Directives / etc]
Hi, I'm trying to create something like the component found here:

https://mui.com/material-ui/react-autocomplete/#githubs-picker

When some element is clicked, a menu pops up with an input and a list of options, and the input value is used as a filter. My first thought was to see if I could do something like this with QSelect, but its field is separate from the menu. This could be done by just sticking an text input in a menu with the list and having the input value filter the list, but I'm hoping to leverage the many features of QSelect like keyboard navigation, multiple options, chips, etc.

Basically, is there a way for QSelect to function with the field and options list rendered within the same element? Is there a better approach to doing this or some other component I'm overlooking? Any advice or suggestions would be appreaciated.

**Comment by VitalickS** [comment]
Hi. I developed some mixin, that can help in your problem. That mixin small and extendable, so you can adopt to any project requirements easily. 

Mixin methods (typescript)
```typescript
{
  filters: <Record<string, string>>reactive({}),
  updateFilter(key: string) {
    const f = this.filters;
    return (val: string, update: (callback: () => void) => void) => update(() => (f[key] = val));
  },
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  getFilter(key: string, filterBy = 'item', caseSensitive = false): (item: any) => boolean {
    const minCharsToFilter = 2;
    let val = this.filters[key];
    if (val === '' || val == null || val === undefined || val.length < minCharsToFilter)
      return () => true;
    if (!caseSensitive) {
      val = val.toUpperCase();
    }
    if (filterBy === '.')
      return (item) => (caseSensitive ? item.includes(val) : item.toUpperCase().includes(val));
    return (item) =>
      caseSensitive ? item[filterBy].includes(val) : item[filterBy].toUpperCase().includes(val);
  },
}
```
Usage example (3 lines of copy/paste code):
```html
  <q-select
    ....
    use-input
    @filter="(val, update) => $api.updateFilter('partner')(val, update)"
    :options="partners.filter($api.getFilter('partner'))"
  />
```
Additionally you can specify '.' to filter by string, and case sensitive or not (with 3rd parameter)
